File: ServiceTest4/app/src/main/java/activitytest/example/com/servicetest4/MyService.java
package activitytest.example.com.servicetest4;

import android.app.Notification;
import android.app.PendingIntent;
import android.app.Service;
import android.content.Intent;
import android.graphics.BitmapFactory;
import android.os.Binder;
import android.os.IBinder;
import android.support.v4.app.NotificationCompat;
import android.util.Log;

public class MyService extends Service {

    private DownloaderBinder mBinder = new DownloaderBinder();

    class DownloaderBinder extends Binder {

        public void startDownload() {
            Log.d("MyService","startDownload executed");
        }

        public int getProgress() {
            Log.d("MyService","getProgress executed");
            return 0;
        }

    }

    public MyService() {
    }

    @Override
    public IBinder onBind(Intent intent) {
        // TODO: Return the communication channel to the service.
        return mBinder;

    }

    @Override
    public void onCreate() {
        super.onCreate();
        Log.d("MyService","onCreate executed");
        Intent intent = new Intent(this,MainActivity.class);
        PendingIntent pi = PendingIntent.getActivity(this,0,intent,0);
        Notification notification = new NotificationCompat.Builder(this,"default")
                .setContentTitle("this is")
                .setContentText("this is text")
                .setWhen(System.currentTimeMillis())
                .setSmallIcon(R.mipmap.ic_launcher)
                .setLargeIcon(BitmapFactory.decodeResource(getResources(),
                        R.mipmap.ic_launcher))
                .setContentIntent(pi)
                .build();
        startForeground(1,notification);
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        Log.d("MyService","onSartCommand executed");
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.d("MyService","onDestroy executed");

    }
}

File: ServiceTest3/app/src/main/java/com/example/servicetest/MyService.kt
package com.example.servicetest

import android.app.DownloadManager
import android.app.Service
import android.content.Intent
import android.os.Binder
import android.os.IBinder
import android.support.annotation.IntDef
import android.util.Log

class MyService : Service() {

    private val mBinder = DownloadBinder()

    internal inner class DownloadBinder : Binder() {

        fun startDownload() {
            Log.d("MyService", "startDownload executed")
        }

        val progress: Int
            get() {
                Log.d("MyService", "getProgress executed")
                return 0
            }

        fun onBind(intent: Intent): IBinder {
            return mBinder
        }

    }

    override fun onBind(intent: Intent): IBinder? {
        // TODO: Return the communication channel to the service.
        throw UnsupportedOperationException("Not yet implemented")
    }

    override fun onCreate() {
        super.onCreate()
        Log.d("MyServie", "oncreate executed")
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        Log.d("MyService", "onStartCommand executed")
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("MyService", "onDestroy executed")
    }
}
